src/hashkit/nc_ketama.c

```diff
@@ -178,6 +178,9 @@ ketama_update(struct server_pool *pool)
             hostlen = snprintf(host, KETAMA_MAX_HOSTLEN, "%.*s-%u",
                                server->name.len, server->name.data,
                                pointer_index - 1);
+            if (hostlen > KETAMA_MAX_HOSTLEN) {
+                hostlen = KETAMA_MAX_HOSTLEN;
```

Comment:
nit: I strongly doubt a use case would exceed the new host length in practice, but if it did, the last byte at `host[KETAMA_MAX_HOSTLEN - 1]` would be `'\0'`, and the implementation as-is would be deterministic it may differ from other implementations for hosts of length much longer than 255

> If the resulting string would be longer than **n-1 characters,** the remaining characters are discarded and not stored, but counted for the value returned by the function.

I can't see that mattering unless OS dns name limits are ever raised from https://www.freesoft.org/CIE/RFC/1035/9.htm

```suggestion
            if (hostlen >= KETAMA_MAX_HOSTLEN) {
                hostlen = KETAMA_MAX_HOSTLEN - 1;
```